### PR TITLE
Bind pflags to default variable fields && Remove short flags as they're not supported by pflags

### PR DESCRIPTION
**Pull request by mayitbeegh**
# TL;DR
This PR attempts to address 2 problems:

1. [PR #69](https://github.com/flyteorg/flytestdlib/pull/69/files) added support for binding pflags to default variable field. The generate code is updated to reflect this change.
2. (Breaking change) short flags are not supported by pflags. They're removed to reduce manual labor and any confusion around the generated code.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
1. `--bind-default-var` is added to all `go:generate pflags` commands.
2. `Go generate` is executed

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1164

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by mayitbeegh** [comment]
Working on fixing TestGetConfig

File: cmd/update/projectconfig_flags.go [hunk 1]
```diff
@@ -41,7 +50,7 @@ func (ProjectConfig) mustMarshalJSON(v json.Marshaler) string {
 // flags is json-name.json-sub-name... etc.
 func (cfg ProjectConfig) GetPFlagSet(prefix string) *pflag.FlagSet {
 	cmdFlags := pflag.NewFlagSet("ProjectConfig", pflag.ExitOnError)
-	cmdFlags.BoolVarP(&(projectConfig.ActivateProject), fmt.Sprintf("%v%v", prefix, "activateProject"), "t", *new(bool), "Activates the project specified as argument.")
-	cmdFlags.BoolVarP(&(projectConfig.ArchiveProject), fmt.Sprintf("%v%v", prefix, "archiveProject"), "a", *new(bool), "Archives the project specified as argument.")
+	cmdFlags.BoolVar(&DefaultProjectConfig.ActivateProject, fmt.Sprintf("%v%v", prefix, "activateProject"), DefaultProjectConfig.ActivateProject, "Activates the project specified as argument.")
+	cmdFlags.BoolVar(&DefaultProjectConfig.ArchiveProject, fmt.Sprintf("%v%v", prefix, "archiveProject"), DefaultProjectConfig.ArchiveProject, "Archives the project specified as argument.")
```
Comment (pmahindrakar-oss) [inline cmd/update/projectconfig_flags.go:54]: Can we also update the documentation removing the usafe of these short flags

https://github.com/flyteorg/flytectl/blob/07f09bc8dd8d29d6ded329d3ebf850ec52b14552/cmd/update/project.go#L37
upto L64.
removed any redundant ones and convert others to use the long version of the flag

File: cmd/config/subcommand/register/filesconfig_flags.go [hunk 2]
```diff
@@ -4,21 +4,58 @@
...
+	cmdFlags.StringVar(&DefaultFilesConfig.Version, fmt.Sprintf("%v%v", prefix, "version"), DefaultFilesConfig.Version, "version of the entity to be registered with flyte.")
+	cmdFlags.BoolVar(&DefaultFilesConfig.ContinueOnError, fmt.Sprintf("%v%v", prefix, "continueOnError"), DefaultFilesConfig.ContinueOnError, "continue on error when registering files.")
+	cmdFlags.BoolVar(&DefaultFilesConfig.Archive, fmt.Sprintf("%v%v", prefix, "archive"), DefaultFilesConfig.Archive, "pass in archive file either an http link or local path.")
+	cmdFlags.StringVar(&DefaultFilesConfig.AssumableIamRole, fmt.Sprintf("%v%v", prefix, "assumableIamRole"), DefaultFilesConfig.AssumableIamRole, " custom assumable iam auth role to register launch plans with.")
+	cmdFlags.StringVar(&DefaultFilesConfig.K8ServiceAccount, fmt.Sprintf("%v%v", prefix, "k8ServiceAccount"), DefaultFilesConfig.K8ServiceAccount, " custom kubernetes service account auth role to register launch plans with.")
+	cmdFlags.StringVar(&DefaultFilesConfig.OutputLocationPrefix, fmt.Sprintf("%v%v", prefix, "outputLocationPrefix"), DefaultFilesConfig.OutputLocationPrefix, " custom output location prefix for offloaded types (files/schemas).")
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/register/filesconfig_flags.go:58]: Similary for this update the documentation in this area to remove the short flag usage.

https://github.com/flyteorg/flytectl/blob/07f09bc8dd8d29d6ded329d3ebf850ec52b14552/cmd/register/files.go#L29

File: cmd/config/subcommand/launchplan/config_flags.go [hunk 3]
```diff
@@ -50,13 +50,12 @@ func (Config) mustMarshalJSON(v json.Marshaler) string {
...
-	cmdFlags.BoolVar((&DefaultConfig.Filter.Asc), fmt.Sprintf("%v%v", prefix, "filter.asc"), false, "Specifies the sorting order. By default flytectl sort result in descending order")
+	cmdFlags.StringVar(&DefaultConfig.ExecFile, fmt.Sprintf("%v%v", prefix, "execFile"), DefaultConfig.ExecFile, "execution file name to be used for generating execution spec of a single launchplan.")
+	cmdFlags.StringVar(&DefaultConfig.Version, fmt.Sprintf("%v%v", prefix, "version"), DefaultConfig.Version, "version of the launchplan to be fetched.")
+	cmdFlags.BoolVar(&DefaultConfig.Latest, fmt.Sprintf("%v%v", prefix, "latest"), DefaultConfig.Latest, " flag to indicate to fetch the latest version,  version flag will be ignored in this case")
+	cmdFlags.StringVar(&DefaultConfig.Filter.FieldSelector, fmt.Sprintf("%v%v", prefix, "filter.field-selector"), DefaultConfig.Filter.FieldSelector, "Specifies the Field selector")
+	cmdFlags.StringVar(&DefaultConfig.Filter.SortBy, fmt.Sprintf("%v%v", prefix, "filter.sort-by"), DefaultConfig.Filter.SortBy, "Specifies which field to sort results ")
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/launchplan/config_flags.go:57]: Not your change , but this format of using - is different than the standard camel case flags we are using.
If you don't mind , can you please fix this and its relevant docs starting from here

https://github.com/flyteorg/flytectl/blob/07f09bc8dd8d29d6ded329d3ebf850ec52b14552/cmd/get/launch_plan.go#L46

**Comment by pmahindrakar-oss** [comment]
Also you can ignore the code coverage issues on this PR , as we are aware of issue where the generated files are not generating the corresponding complete coverage.
@EngHabu suggested to use flag in the generator to exclude the function generation which is not required in the consumer which might help with this issue or we can add the correponding fix in the code generator to generate full code coverage for the uncovered code.

In interim we will be able disregard the coverage check generated by codecov to merge such changes.

File: cmd/config/subcommand/launchplan/config_flags.go (hunk 3 above)
Comment (pmahindrakar-oss) [reply cmd/config/subcommand/launchplan/config_flags.go:57]: https://github.com/flyteorg/flytectl/blob/master/docs/CONTRIBUTING.md
cc : @evalsocket

File: cmd/update/projectconfig_flags.go (hunk 1 above)
Comment (mayitbeegh) [reply cmd/update/projectconfig_flags.go:54]: Removed

File: cmd/config/subcommand/launchplan/config_flags.go (hunk 3 above)
Comment (mayitbeegh) [reply cmd/config/subcommand/launchplan/config_flags.go]: Fixed

File: cmd/config/subcommand/register/filesconfig_flags.go (hunk 2 above)
Comment (mayitbeegh) [reply cmd/config/subcommand/register/filesconfig_flags.go:58]: Removed. Anything needs to be done about the .rst files?

Comment (pmahindrakar-oss) [reply cmd/config/subcommand/register/filesconfig_flags.go:58]: The rst get generated  from the section that you edited for this automatically.
make gendocs is the endpoint for generating the docs from within the docs folder.

flyte-bot takes care of that if it finds something new has been added and creates a PR for it .
You can do it aswell  to override the bots behavior

File: cmd/config/subcommand/launchplan/config_flags.go (hunk 3 above)
Comment (pmahindrakar-oss) [reply cmd/config/subcommand/launchplan/config_flags.go]: thank you

File: cmd/config/subcommand/register/filesconfig_flags.go (hunk 2 above)
Comment (mayitbeegh) [reply cmd/config/subcommand/register/filesconfig_flags.go:58]: Got it.

File: pkg/filters/type.go [hunk 4]
```diff
@@ -10,8 +10,8 @@ var (
 )
 
 type Filters struct {
-	FieldSelector string `json:"field-selector" pflag:",Specifies the Field selector"`
-	SortBy        string `json:"sort-by" pflag:",Specifies which field to sort results "`
+	FieldSelector string `json:"fieldSelector" pflag:",Specifies the Field selector"`
+	SortBy        string `json:"sortBy" pflag:",Specifies which field to sort results "`
 	// TODO: Support paginated queries
```
Comment (pmahindrakar-oss) [inline pkg/filters/type.go:15]: With this change , there are docs references as well which we would need to change . 
Can you check all references of field-selector and sort-by in go files and replace them.
eg : in get/launch_plan.go

If you do make gendocs , it will take care to fix the .rst files aswell.

Comment (mayitbeegh) [reply pkg/filters/type.go:15]: Fixed

**Review by pmahindrakar-oss — APPROVED** [review]
LGTM
